### PR TITLE
Fix for Issue #46

### LINKED ISSUE #46: ObjectiveControlPoint doesn't reflect Team changes due to new Team/Player system
If a Faction has a Quest Objective to capture a particular Control Point, and that Faction joins the Team that controls that Control Point, the Objective should be completed. Right now that doesn't happen. Here are all the cases to cover:
 * Complete the objective when joining a Team that has the Control Point
 * Complete the objective when a Faction joins _your_ Team and they have the Control Point
 * Incomplete the objective when leaving a Team that has the Control Point
 
 This issue may also apply to ObjectiveControlLegend.

### PR DESCRIPTION
**Pull request by SavanticIO**
Closes #46.

I'm not 100% sure if these changes follow the new Team/Player system or not.

i ran tests for the following scenarios;

- Complete the objective when joining a Team that has the Control Point
- Complete the objective when a Faction joins your Team and they have the Control Point
- Incomplete the objective when leaving a Team that has the Control Point
- Incomplete the objective a Faction leaves your Team and they have the Control Point

### CODE REVIEW

File: src/MacroTools/QuestSystem/UtilityStructs/ObjectiveControlPoint.cs [hunk 1]
```diff
@@ -16,17 +16,19 @@ public ObjectiveControlPoint(ControlPoint target)
...
     internal override void OnAdd(Faction whichFaction)
     {
-      if (IsPlayerOnSameTeamAsAnyEligibleFaction(_target.Unit.OwningPlayer()))
-        Progress = QuestProgress.Complete;
+      Progress = IsPlayerOnSameTeamAsAnyEligibleFaction(_target.Unit.OwningPlayer())
+        ? QuestProgress.Complete
+        : QuestProgress.Incomplete;
 
-      whichFaction.JoinedTeam += OnFactionTeamJoin;
+      //whichFaction.JoinedTeam += OnFactionTeamJoin;
```
Comment (YakaryBovine) [inline src/MacroTools/QuestSystem/UtilityStructs/ObjectiveControlPoint.cs:31]: Might as well remove this.

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Looks good to me. Just one tiny thing. I haven't tested it thoroughly myself but it sounds like you've got that covered. It looks correct.

File: src/MacroTools/QuestSystem/UtilityStructs/ObjectiveControlPoint.cs (hunk 1 above)
Comment (SavanticIO) [reply src/MacroTools/QuestSystem/UtilityStructs/ObjectiveControlPoint.cs]: Removed

**Review by YakaryBovine — APPROVED** [review]
Awesome, well done. This must have been at least a little challenging. And thanks for being thorough and dealing with ControlLegend as well.

**Pull request merged**